tensorflow_datasets/core/as_dataframe_test.py

```diff
@@ -37,6 +37,53 @@ def _as_df(ds_name: str, **kwargs) -> pandas.DataFrame:
   return df
 
 
+def test_flatten_with_path():
+  """Test that the flatten function works as expected"""
+  from from tensorflow_datasets.core.as_dataframe import _flatten_with_path
+  assert list(_flatten_with_path('value')) == [([], 'value')]
+  assert list(_flatten_with_path({'key': 'value'})) == [('key', 'value')]
+  assert list(_flatten_with_path({'key1': 'value', 'key2': 'value2'})) == [
+    ('key1', 'value'),
+    ('key2', 'value2'),
+  ]
+  complex_dict = {
+    'key': 'value',
+    'nested': {
+      'sub_key': 'subvalue',
+      'sub_nested': {
+        'subsubkey1': 'subsubvalue1',
+        'subsubkey2': 'subsubvalue2',
+      },
+      'key2': 'value2',
+    }
+  }
+  assert list(_flatten_with_path(complex_dict)) == [
+    (['key'], 'value'),
+    (['key2'], 'value2'),
+    (['nested', 'subkey'], 'subvalue'),
+    (['nested', 'sub_nested', 'subsubkey1'], 'subsubvalue1'),
+    (['nested', 'sub_nested', 'subsubkey2'], 'subsubvalue2'),
+  ]
```

Comment:
I'll also add a test which check that `flatten_with_path()` sorted order is consistent with `tf.nest.flatten`.

Reply:
Sure, can do. To be sure: Consistency with tf.nest.flatten or _flatten (if that should be kept)?

Reply:
Yes, in theory, it should be the case, as `tf.nest.flatten` also sort the keys. But it's best to guarantee this with a test.

Reply:
Looks like TF has `flatten_with_tuple_paths_up_to` which is mostly like `flatten_with_path`. I'll see if that can be used too.

Reply:
> Looks like TF has flatten_with_tuple_paths_up_to which is mostly like flatten_with_path. I'll see if that can be used too.

This would have been  nice, but as far as I know, this was part of tf.contrib, which is deprecated and not included in TF anymore.

As far as I know, this is the current symbol for the TF API: https://www.tensorflow.org/api_docs/python/tf/nest ?